fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateTableCommand.java

```diff
@@ -0,0 +1,178 @@
+
+    private static final ShowResultSetMetaData VIEW_META_DATA =
+            ShowResultSetMetaData.builder()
+            .addColumn(new Column("View", ScalarType.createVarchar(20)))
+            .addColumn(new Column("Create View", ScalarType.createVarchar(30)))
+            .addColumn(new Column("character_set_client", ScalarType.createVarchar(30)))
+            .addColumn(new Column("collation_connection", ScalarType.createVarchar(30)))
+            .build();
+
+    private static final ShowResultSetMetaData MATERIALIZED_VIEW_META_DATA =
+            ShowResultSetMetaData.builder()
+            .addColumn(new Column("Materialized View", ScalarType.createVarchar(20)))
+            .addColumn(new Column("Create Materialized View", ScalarType.createVarchar(30)))
+            .build();
+
+    private final TableNameInfo tblNameInfo;
+    private final boolean isBrief;
+
+    public ShowCreateTableCommand(TableNameInfo tableNameInfo, boolean isBrief) {
+        super(PlanType.SHOW_CREATE_TABLE_COMMAND);
+        this.tblNameInfo = tableNameInfo;
+        this.isBrief = isBrief;
+    }
+
+    private void validate(ConnectContext ctx) throws AnalysisException {
+        tblNameInfo.analyze(ctx);
+
+        TableIf tableIf = Env.getCurrentEnv().getCatalogMgr()
+                .getCatalogOrAnalysisException(tblNameInfo.getCtl())
+                .getDbOrAnalysisException(tblNameInfo.getDb()).getTableOrAnalysisException(tblNameInfo.getTbl());
+
+        if (tableIf instanceof MTMV) {
+            ErrorReport.reportAnalysisException("not support async materialized view, "
+                    + "please use `show create materialized view`");
+        }
+
+        PrivPredicate wanted;
+        if (tableIf instanceof View) {
+            wanted = PrivPredicate.SHOW_VIEW;
+        } else {
+            wanted = PrivPredicate.SHOW;
+        }
+
+        if (!Env.getCurrentEnv().getAccessManager().checkTblPriv(ConnectContext.get(),
+                tblNameInfo.getCtl(), tblNameInfo.getDb(), tblNameInfo.getTbl(), wanted)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_TABLEACCESS_DENIED_ERROR, "SHOW CREATE TABLE",
+                                                ConnectContext.get().getQualifiedUser(),
+                                                ConnectContext.get().getRemoteIP(),
+                                                tblNameInfo.getDb() + ": " + tblNameInfo.getTbl());
+        }
+    }
+
+    public boolean isBrief() {
+        return isBrief;
+    }
+
+    @Override
+    public <R, C> R accept(PlanVisitor<R, C> visitor, C context) {
+        return visitor.visitShowCreateTableCommand(this, context);
+    }
+
+    @Override
+    public ShowResultSet doRun(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        validate(ctx);
+        // If dbName is not provided, use the default database from the context
+        String effectiveDbName = (tblNameInfo.getDb() == null || tblNameInfo.getDb().isEmpty())
```

Comment:
please refer to handleShowCreateTable() in ShowExecutor.java. The old code just needs some minor changes, it's easier to reivew. thanks.

Reply:
done - please review